### PR TITLE
Fix Bugzilla 17148 - Copying from const(void)[] to void[] breaks immu…

### PR DESCRIPTION
**Pull request by ntrel**
…table

Disallow copying if source type won't implicitly convert to destination type, ~~for the next edition~~ with fixImmutableConv preview.

### CODE REVIEW

**Comment by ntrel** [comment]
Phobos won't build yet with the new error enabled (I'll make a pull). It looks like there is no CI to test building dmd, druntime, phobos with the next edition.

**Review by thewilsonator — APPROVED** [review]

**Comment by thewilsonator** [comment]
Dependant PR merged, please rebase to retrigger CI

**Comment by thewilsonator** [comment]
Merge conflict.

File: changelog/dmd.copying-to-void-arrays.dd [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+If `const(void)[]` data contains tail `const` pointers, copying to `void[]`
+can subsequently violate `const` data:
+---
+void f(int*[] a, const int*[] b)
+{
+	void[] va = a;
+	const void[] vb = b;
+	va[] = vb[]; // fills `a` with pointers to const
+	*a[0] = 0; // const data mutated
```
Comment (thewilsonator) [inline changelog/dmd.copying-to-void-arrays.dd:11]: DAutoTest fails with:
```
Searching for tabs
./changelog/pending.dd:33:	void[] va = a;
./changelog/pending.dd:34:	const void[] vb = b;
./changelog/pending.dd:35:	va[] = vb[]; // fills `a` with pointers to const
./changelog/pending.dd:36:	*a[0] = 0; // const data mutated
```

**Pull request merged**